### PR TITLE
Add distr predictions to surv.glmnet learners

### LINKED ISSUE #296: Add `distr` survival prediction in `surv.glmnet` learners
We can easily I think include the `distr` prediction (with Breslow estimator) after `glmnet` implemented [glmnet:::survfit.coxnet](https://www.rdocumentation.org/packages/glmnet/versions/4.1/topics/survfit.coxnet)

Same logic as is done in coxPH => https://github.com/mlr-org/mlr3proba/blob/main/R/LearnerSurvCoxPH.R#L65

After this, both `surv.glmnet` and `surv.cv_glmnet` would not to be wrapped in a pipeline to get the `distr` predictions (as is done now in the corresponding mlr3book [section](https://mlr3book.mlr-org.com/chapters/chapter13/beyond_regression_and_classification.html#explicit-composition-and-pipelines))

### PR DESCRIPTION
**Pull request by bblodfon**
* fixes #296

### CODE REVIEW

**Comment by bblodfon** [comment]
@RaphaelS1 or @sebffischer can you review?

Summary:
- I had to include the training data (`x`, `y`) in the `model` slot, `distr` via `survfit` cannot be computed otherwise. Just like `surv.bart`, `learner$model` is now a list.
- `s` parameter had an `upper = 1` in one of the learners which was wrong (`s` is the `lambda` for prediction so it is just positive)
- Tested the `weights` and `offset`/`newoffset` parameters, works as expected
- Added the `ctype` and `stype` which can change (very slightly) the results of the `distr` predictions (as is in `surv.coxph`)

PS: @be-marc the functions related to `lambda` and `s` were very well done ;)

**Review by RaphaelS1 — COMMENTED** [review]
LGTM but only gave the code a quick glance and didn't test (so I'll let Sebastian approve)

File: R/learner_glmnet_surv_glmnet.R [hunk 1]
```diff
@@ -10,7 +10,18 @@
 #' - `family` is set to `"cox"` and cannot be changed.
 #'
 #' @details
-#' Caution: This learner is different to learners calling [glmnet::cv.glmnet()]
+#' This learner returns two prediction types:
```
Comment (sebffischer) [inline R/learner_glmnet_surv_glmnet.R:13]: if you have the same documentation twice you can also use a template or something like the `@inherit` / `@inheritSection` tags from roxygen

File: R/learner_glmnet_surv_cv_glmnet.R [hunk 2]
```diff
@@ -108,16 +123,31 @@ delayedAssign(
           pv$weights = task$weights$weight
         }
 
-        glmnet_invoke(data, target, pv, cv = TRUE)
+        list(
+          model = glmnet_invoke(data, target, pv, cv = TRUE),
+          # need these for distr prediction
+          x = data,
```
Comment (sebffischer) [inline R/learner_glmnet_surv_cv_glmnet.R:129]: It needs to be possible to disable this, we don't always want to store the complete training data.
This changes the size of the stored model by a pretty big factor I think when we have large datasets and should be something to be considered. 

I suggest the following:
there should be a parameter that turns the saving of the model (and hence whether a `distr` prediction can be made) on and off. This parameter should be `FALSE` per default. I think we should really strive to avoid very large objects in the `$model` slot.

Comment (sebffischer) [reply R/learner_glmnet_surv_cv_glmnet.R:129]: we then also need to throw a proper error when the `distr` predict type is requested during `$.predict` but the required information is not available.

File: R/learner_glmnet_surv_cv_glmnet.R [hunk 3]
```diff
@@ -53,15 +65,16 @@ delayedAssign(
           mxitnr               = p_int(1L, default = 25L, tags = "train"),
           nfolds               = p_int(3L, default = 10L, tags = "train"),
           nlambda              = p_int(1L, default = 100L, tags = "train"),
-          offset               = p_uty(default = NULL, tags = "train"),
+          offset               = p_uty(default = NULL, tags = c("train", "predict")),
+          newoffset            = p_uty(tags = "predict"),
```
Comment (sebffischer) [inline R/learner_glmnet_surv_cv_glmnet.R:69]: I don't understand this, why does `offset` now have two tags but there is also a new `newoffset` parameter?

File: R/learner_glmnet_surv_glmnet.R [hunk 4]
```diff
@@ -70,7 +81,7 @@ delayedAssign(
           mxitnr           = p_int(1L, default = 25L, tags = "train"),
           newoffset        = p_uty(tags = "predict"),
           nlambda          = p_int(1L, default = 100L, tags = "train"),
-          offset           = p_uty(default = NULL, tags = "train"),
+          offset           = p_uty(default = NULL, tags = c("train", "predict")),
```
Comment (sebffischer) [inline R/learner_glmnet_surv_glmnet.R:84]: here there is no `newoffset` parameter.

File: R/learner_glmnet_surv_glmnet.R (hunk 1 above)
Comment (sebffischer) [reply R/learner_glmnet_surv_glmnet.R:13]: --> this requires this information to be used to its own section

**Review by sebffischer — CHANGES_REQUESTED** [review]
Hey John, thanks a lot for the PR! 
I added my comments. 

Please also add some small sanity tests to verify that the added prediction type is working.

File: R/learner_glmnet_surv_glmnet.R (hunk 1 above)
Comment (bblodfon) [reply R/learner_glmnet_surv_glmnet.R:13]: They are a bit different between the two learners, but maybe you know a trick or two on how to do it (to discuss today)

File: R/learner_glmnet_surv_cv_glmnet.R (hunk 3 above)
Comment (bblodfon) [reply R/learner_glmnet_surv_cv_glmnet.R:69]: `offset` is needed in the `.predict` method and it says so also in the [documentation](https://www.rdocumentation.org/packages/glmnet/versions/4.1/topics/survfit.coxnet)

`offset` is a vector the size of the train set and if used during training, it is also needed during prediction along with a `newoffset` (vector the size of the test set).

File: R/learner_glmnet_surv_glmnet.R (hunk 4 above)
Comment (bblodfon) [reply R/learner_glmnet_surv_glmnet.R:84]: See line 82 :)

**Comment by bblodfon** [comment]
@sebffischer ready to merge

**Comment by bblodfon** [comment]
I need to run `document()` once more I think but I have an issue with it (see Mattermost)

**Review by sebffischer — APPROVED** [review]
